packages/lit-html/README.md

```diff
@@ -1,14 +1,140 @@
+  * Custom `templateFactory`, `TemplateProcessor`, and custom tag functions are no
+    longer supported
+
+See the full [changelog](CHANGELOG.md#200---unreleased) for more details on
+these and other minor breaking changes.
+
+## 🚨 Known issues/limitations
+
+* **Browser support**: This pre-release should run on modern browsers, however a
+  change to factor legacy browser support (IE11, etc.) into an opt-in package is
+  ongoing. As such, this release will not run on some older browsers. This is a
+  temporary state.
+* **Limited directive implementation**: The following directives are not yet
+  implemented. This is a temporary state:
+  * `asyncAppend`
+  * `asyncReplace`
+  * `templateContent`
+  * `until`
+
+## 🚨 Migrating directives
+
+While the API for _using_ directives should be 100% backward-compatible with
+1.x, there is a breaking change to how custom directives are _authored_. The API
+change improves ergonomics around making stateful directives while providing a
+clear pattern for SSR-compatible directives: only `render` will be called on the
+server, while `update` will not be.
+
+**⚠️ WARNING: The directive and part API changes are experimental and subject to
+change in future pre-releases.**
+
+<details>
+<summary>Expand here for details on migrating directives.</summary>
+
+### Overview of directive API changes
+| | 1.x API | 2.0 API |
+|-|-----|-----|
+| Code idiom for directive | function that takes directive arguments, and returns function that takes `part` and returns value | class with `update` & `render` methods which accept directive arguments |
+| Where to do declarative rendering | pass value to `part.setValue()` | return value from `render()` method |
+| Where to do imperative DOM/part manipulation | directive function | `update()` method |
+| Where state is stored between renders | `WeakMap` keyed on `part` | class instance fields |
+| How part validation is done | `instanceof` check on `part` in every render | `part.type` check in constructor
+
+### Example directive migration
+Below is an example of a fictional directive, and how to migrate it to the new
+API:
+
+1.x Directive API:
+```js
+import {directive, NodePart, html} from 'lit-html';
+
+// State stored in WeakMap
+const previousState = new WeakMap();
+
+// Functional-based directive API
+export const renderCounter = directive((initialValue) => (part) => {
+  // When necessary, validate part type each render using `instanceof`
+  if (!(part instanceof NodePart)) {
+    throw new Error('renderCounter only supports NodePart');
+  }
+  // Retrieve value from previous state
+  let value = previousState.get(part);
+  // Update state
+  if (previous === undefined) {
+    value = initialValue;
+  } else {
+    value++;
+  }
+  // Store state
+  previousState.set(part, value);
+  // Update part with new rendering
+  part.setValue(html`<p>${value}</p>`);
+});
+```
+
+2.0 Directive API:
+```js
+import {directive, Directive, NODE_PART} from 'lit-html';
```

Comment:
You still need to import `html` here, right?

Reply:
Yes, thanks!